src/main/java/org/springframework/integration/aws/support/S3Session.java

```diff
@@ -236,7 +236,7 @@ public Object getClientInstance() {
 
 	private String[] splitPathToBucketAndKey(String path) {
 		Assert.hasText(path, "'path' must not be empty String.");
-		String[] bucketKey = path.split("/");
+		String[] bucketKey = path.split("/", 2);
```

Comment:
Yes, that would be great to have a comprehensive PR from contributor. 
Otherwise I will wait until tomorrow and merge it with my sane.